test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -7616,6 +7616,29 @@ def forward(self, x):
         x = torch.tensor([[1, 2, float("inf")], [2, float("nan"), float("inf")]])
         self.run_test(M(), (x, ))
 
+    @skipIfUnsupportedMinOpsetVersion(10)  # ONNX IsNaN, IsInf op is added in opset 9, 10 respectively.
+    def test_nan_to_num(self):
+        class NoParams(torch.nn.Module):
+            def forward(self, x):
+                return x.nan_to_num()
+
+        x = torch.tensor([[1, 2, float("inf")], [2, float("nan"), -float("inf")]])
+        self.run_test(NoParams(), (x, ))
+
+        class WithParams(torch.nn.Module):
+            def forward(self, x):
+                return x.nan_to_num(nan=2.3, posinf=4.5, neginf=6.7)
+
+        x = torch.tensor([[1, 2, float("inf")], [2, float("nan"), -float("inf")]])
+        self.run_test(WithParams(), (x, ))
+
+        class IntModel(torch.nn.Module):
```

Comment:
IIUC, this is identical to NoParams. If so, please re-use that class and pass in `test_with_inputs`.

Reply:
IIUC the test cases differ only in the inputs passed to the model, which you can vary using the `test_with_inputs` argument to `self.run_test`, and you don't need two classes (just use `NoParams` ).

Reply:
I'm currently trying 
```
x = torch.tensor([[1, 2, float("inf")], [2, float("nan"), -float("inf")]])
        x1 = torch.tensor([[1, 2, 4], [8, 16, 32]])
        self.run_test(NoParams(), (x, ),
                      test_with_inputs=[(x, ), (x1, )])
                    ```
                   
And run into the issue
```
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Unexpected input data type. Actual: (tensor(int64)) , expected: (tensor(float))
```

Reply:
Sure, two calls to `run_test(NoParams()` seems fine.